askama_derive/src/generator.rs

```diff
@@ -727,6 +727,42 @@ impl<'a> Generator<'a> {
                 args.len()
             )));
         }
+        let mut output_args = Vec::new();
```

Comment:
This seems pretty messy. I'd like to have a comment here that explains the high-level logic for handling invocations that have both named and unnamed arguments, which seems like the tricky case here. Then, I think there should be one pass over the arguments that both splits the arguments in `unnamed_args` and `named_args` (I think we should be able to do this without cloning?) and checks whether all the names are acceptable.

Reply:
This is still iterating `args` twice: once for `args.iter().any()` and once for `args.iter().partition()`. I'm pretty sure this isn't actually necessary?

Reply:
I can simplify it even further indeed. I'll do that.

Reply:
My idea couldn't work actually (sorry for the noise). The problem is that I first need to **ensure** that all named arguments actually have an equivalent in the macro, otherwise they'll be passed as non-named arguments and the `remove` call would fail. However, a first improvement that could be done would be to move this loop after we split both kind of arguments so that we only iterate on named arguments.

Reply:
Since named arguments are always last, it allowed to remove the first loop and to instead just check if the last element is a named argument.